csrc/env.h

```diff
@@ -50,8 +51,7 @@ class java_ex {
         jthrowable m_java_exception;
 
         const char *m_java_classname;
-        const std::string m_message;
-        const char *m_message_cstr;
```

Comment:
We didn't previously have a maximum message length. While this works why not just always store the message in `m_message` which is an actual C++ object and thus will properly manage its memory as we need?

Reply:
Done. Switched this exception handling to use `std::string` rather than plain C strings.